gax/src/main/java/com/google/api/gax/batching/Batcher.java

```diff
@@ -79,4 +81,10 @@
    */
   @Override
   void close() throws InterruptedException;
+
+  /**
+   * Closes this Batcher by preventing new elements from being added and send outstanding elements.
+   * The returned future will be resolved when the last element completes
+   */
+  ApiFuture<Void> closeAsync();
```

Comment:
This is an API breaking change.

Reply:
Batcher interface was meant to be InternalExtensionOnly and it is a bug that it was not. The api is marked as BetaApi, which should permit us to make this change

Reply:
I'm ok with adding it if we can validate somehow, that at least none of the google libraries is currently extending it (I mean manual or autogenerated). Most probably they don't.